### DISCUSSION TITLE
RfC - django CMS Association - streamlining membership processes with Github Sponsors

### DISCUSSION THREAD

**Discussion by macolo** [django-cms/django-cms#7839] [General]
## Intro
- we have recently joined the Github Sponsors program and have received a sponsor page in our org's github profile: https://github.com/sponsors/django-cms

### General setup steps
- [x] create a welcome message per tier on https://github.com/sponsors/django-cms/dashboard/tiers/56747/edit

### credit card payments
- the dCA has been offering cc payments for individual memberships to avoid the hassle of international bank transfers for abroad members.
- until now, the dCA used a borrowed stripe account from one of the founding member's company.
- With Github Sponsors program, a stripe account in the name of the dCA has been created
- [x] we should shift all credit card payments to Github Sponsors and phase out of payments with the legacy stripe account

### Annual payments with Github Sponsors
- also see https://github.com/orgs/community/discussions/11568
- Github Sponsors payments respect the user's billing settings. A github user can switch to annual payments globally for their account at https://github.com/settings/billing/summary 
- this can be done for org and individual accounts separately
- here is github's howto: https://docs.github.com/en/billing/managing-your-github-billing-settings/changing-the-duration-of-your-billing-cycle#changing-the-duration-of-your-personal-accounts-billing-cycle
- after upgrading to annual billing cycle, a user who for example selects the USD 2 monthly tier will be billed 24 USD once a year.
- [x] We should let users know on https://github.com/sponsors/django-cms and in any signup emails from the member onboarding process

### Automating the Onboarding process for individual members
- using Github Sponsor, we now have the opportuntiy to automate the onboarding process for individual members
- current manual process:
   - new member fills in the email form at https://www.django-cms.org/en/sign-up/
   - Growth manager sends a welcome email
   - Growth manager creates membership application and sends link to sign the statutes
   - Growth manager creates new entry in membership spreadsheet
   - Accountant creates a bexio invoice or stripe payment link (?) and sends via email to member
   - Accountant repeats the previous step annually (360*x days after signup date) 
- New (semi-) automated process:
   - user is directed to https://github.com/sponsors/django-cms from https://www.django-cms.org/en/memberships/ supporter box 
   - optionally, user switches github account to annual billing cycle
   - user sees the dCA membership tier and is informed that by selecting the dCA membership tier they accept the dCA statutes
   - user goes through the tier signup steps with Github Sponsors and triggers email to office@ 
   - Growth manager adds user to the members spreadsheet on google drive
   - Henceforth, Github Sponsors charges the individual members credit card monthly or yearly
- [x] remove individual membership from the dropdown menu at https://www.django-cms.org/en/sign-up/
- [x] remove individual users list from https://www.django-cms.org/en/our-members/ (link instead to supporter list on github.com)

### Github Sponsors buttons on the project level
- https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/displaying-a-sponsor-button-in-your-repository
- a yml file has to be committed to each project separately. This will have to be done respecting collaboration policies of each individual project (i.e. by creating a PR etc.)
- where necessary, a corresponding ticket can be created.

**Comment by fsbraun** [comment]
We should add the sponsor button to all endorsed repos. 👍

**Comment by fsbraun** [comment]
Does the supporter get listed on the web page as promised automatically, or is that a separate step?

**Reply by macolo** [reply]
True, it is a manual step so far. I updated the RfC in order to propose a simpler solution:
> remove individual users list from https://www.django-cms.org/en/our-members/ (link instead to supporter list on github.com)

**Reply by marksweb** [reply]
I bet it can be automated from the API

**Comment by macolo** [comment]
In the end, we did the following:
- restructured the [supporters page](https://www.django-cms.org/en/community/django-cms-supporters)
- we keep the individual membership, however membership fee for individuals is to be done through Github Sponsors
